init.c

```diff
@@ -325,6 +325,7 @@ void free_threads_shm(void)
 	cur_segment = 0;
 }
 
+#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
```

Comment:
Maybe move the ifndef into the body so you don't need it in two places?

Reply:
Good idea